src/coffeescript.coffee

```diff
@@ -124,7 +124,8 @@ exports.compile = compile = withPrettyErrors (code, options = {}) ->
     ast.loc.start = ast.program.loc.start = {line: 1, column: 0}
     ast.loc.end.line = ast.program.loc.end.line = sourceCodeNumberOfLines
     ast.loc.end.column = ast.program.loc.end.column = sourceCodeLastLine.length
-    return ast
+    return ast unless options.withTokens
```

Comment:
Is this adding a new field to the `compile` options object, `withTokens`? Can we maybe make `ast` take an object rather than be a Boolean, and AST options can go in there? Or can we just always return tokens?

The external API is deliberately kept very minimal, to lessen the load on build tool plugins that wrap CoffeeScript.

Reply:
Ok updated so that it just automatically attaches `tokens` to the `ast` object (ie the root AST node)

If we end up having a reason to not *always* attach the tokens, then ya that seems reasonable to make it accept eg `ast: {tokens: yes}` instead of just `ast: yes`

See as a reference the [Babel parser API](https://babeljs.io/docs/en/babel-parser), which will optionally attach `tokens` to the returned root AST node if you pass a `tokens` option when parsing

Reply:
I assume the performance difference is miniscule, since the `tokens` object already exists? So it’s really a question of whether we want to return a bloated AST or not. I would err on the side of returning everything, at least at first, and maybe later on add options for slimming it down (i.e. no tokens, no comments, etc.)?

We might very well want to provide more AST-related options eventually, for example to return only a Babel-compatible AST or an AST without CoffeeScript-specific constructs like `Range`. I think it would be best to get the simplest version out the door first before we go down that road, so that we can start to see at least one or two use cases in action before we add more features.

Reply:
👍 that all makes sense to me